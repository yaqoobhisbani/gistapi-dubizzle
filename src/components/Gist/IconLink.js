import React from "react";
import styled from "styled-components";
import Octicon from "react-octicon";
import { colors } from "../../theme/colors";

const IconLink = ({ icon, label, link, style }) => {
  return (
    <Wrapper style={style}>
      <Octicon name={icon} style={iconStyle} />
      <Label href={link}>{label}</Label>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

const Label = styled.a`
  font-size: 16px;
  color: ${(props) => props.theme.primary};
  text-decoration: none;
  cursor: pointer;
  margin-left: 4px;
`;

const iconStyle = {
  color: colors.primary,
  fontSize: 16,
  fontWeight: "500",
};

export default IconLink;
