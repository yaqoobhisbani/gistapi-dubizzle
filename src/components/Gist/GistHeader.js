import React from "react";
import styled from "styled-components";
import IconLink from "./IconLink";

const getFilesLabel = (count) => {
  if (count == 0) return count + " " + "Files";
  if (count == 1) return count + " " + "File";
  if (count > 1) return count + " " + "Files";
};

const GistHeader = ({
  avatarURL,
  login,
  loginURL,
  commentsURL,
  forksURL,
  fileCount,
}) => {
  return (
    <Header>
      <Person>
        <Photo src={avatarURL} />
        <NameLink href={loginURL}>{login}</NameLink>
      </Person>

      <Row>
        <IconLink
          label={getFilesLabel(fileCount)}
          icon="code"
          style={iconLink}
        />
        <IconLink
          label="Forks"
          icon="repo-forked"
          link={forksURL}
          style={iconLink}
        />
        <IconLink
          label="Comments"
          icon="comment"
          link={commentsURL}
          style={iconLink}
        />
        <IconLink label="Stars" icon="star" />
      </Row>
    </Header>
  );
};

const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 16px;
`;

const Person = styled.div`
  display: flex;
  align-items: center;
`;

const Photo = styled.img`
  width: 36px;
  height: 36px;
  border-radius: 500px;
`;

const NameLink = styled.a`
  font-size: 16px;
  color: ${(props) => props.theme.primary};
  text-decoration: none;
  cursor: pointer;
  margin-left: 8px;
`;

const Row = styled.div`
  display: flex;
  align-items: center;
`;

const iconLink = { marginRight: 16 };

export default GistHeader;
