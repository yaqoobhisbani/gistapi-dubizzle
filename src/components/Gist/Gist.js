import React from "react";
import styled from "styled-components";
import Date from "./Date/Date";
import GistHeader from "./GistHeader/GistHeader";
import IconLink from "./IconLink/IconLink";

const Gist = ({ gist }) => {
  return (
    <Wrapper>
      <GistHeader
        avatarURL={gist?.owner?.avatar_url}
        login={gist?.owner?.login}
        commentsURL={gist?.comments_url}
        forksURL={gist?.forks_url}
        loginURL={gist?.owner?.html_url}
        fileCount={Object.keys(gist.files).length}
      />

      <Row>
        <Date label="Created at: " date={gist?.created_at} />
        <Date
          label="Last updated: "
          date={gist?.updated_at}
          style={styles.date}
        />
      </Row>

      <Description>{gist?.description}</Description>

      <FilesRow>
        {Object.values(gist.files).map((file, index) => (
          <IconLink
            label={file.filename}
            icon="file-text"
            link={file.raw_url}
            key={`${file.filename}_${index}`}
            style={styles.file}
          />
        ))}
      </FilesRow>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 10px;
  border-bottom: 1px solid ${(props) => props.theme.border};
  padding: 16px 0;
`;

const Row = styled.div`
  display: flex;
  align-items: center;
`;

const FilesRow = styled(Row)`
  flex-wrap: wrap;
`;

const Description = styled.p`
  color: ${(props) => props.theme.darkgray};
  font-size: 16px;
  padding: 16px 0;
`;

const styles = {
  file: { marginRight: 16 },
  date: { marginLeft: 16 },
};

export default React.memo(Gist);
