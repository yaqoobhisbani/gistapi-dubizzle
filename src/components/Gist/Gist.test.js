import React from "react";
import { render } from "@testing-library/react";
import Gist from "./Gist";

describe("Gist", () => {
  test("matches snapshot", () => {
    const gist = {
      url: "https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a",
      forks_url:
        "https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a/forks",
      commits_url:
        "https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a/commits",
      id: "840386720b6a72d9e60dff8922f9747a",
      node_id: "G_kwDOAYRM6toAIDg0MDM4NjcyMGI2YTcyZDllNjBkZmY4OTIyZjk3NDdh",
      git_pull_url:
        "https://gist.github.com/840386720b6a72d9e60dff8922f9747a.git",
      git_push_url:
        "https://gist.github.com/840386720b6a72d9e60dff8922f9747a.git",
      html_url:
        "https://gist.github.com/GrahamcOfBorg/840386720b6a72d9e60dff8922f9747a",
      files: "{Potential Maintainers: {…}}",
      public: true,
      created_at: "2023-03-30T21:44:05Z",
      updated_at: "2023-03-30T21:44:05Z",
      description: "test",
      comments: 0,
      user: null,
      comments_url:
        "https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a/comments",
      owner: '{avatar_url: "https://avatars.githubusercontent.com…}',
      truncated: false,
    };

    const { container, getByText } = render(<Gist gist={gist} />);
    const textElement = getByText(gist.description);
    expect(textElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
