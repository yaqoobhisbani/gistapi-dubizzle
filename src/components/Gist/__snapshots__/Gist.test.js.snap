// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gist matches snapshot 1`] = `
<div>
  <div
    class="sc-fsQiph hGyyWv"
  >
    <div
      class="sc-hLseeU BbBzW"
    >
      <div
        class="sc-eDDNvR idlsVM"
      >
        <img
          class="sc-jTrPJq dYZuXy"
        />
        <a
          class="sc-gLDzan dAgnKx"
        />
      </div>
      <div
        class="sc-iAEyYk fIvPjx"
      >
        <div
          class="sc-gueYoa cHbybZ"
          style="margin-right: 16px;"
        >
          <span
            class="octicon octicon-code"
            data-testid="icon"
            style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
          />
          <a
            class="sc-dmqHEX lmtyIK"
          >
            28 Files
          </a>
        </div>
        <div
          class="sc-gueYoa cHbybZ"
          style="margin-right: 16px;"
        >
          <span
            class="octicon octicon-repo-forked"
            data-testid="icon"
            style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
          />
          <a
            class="sc-dmqHEX lmtyIK"
            href="https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a/forks"
          >
            Forks
          </a>
        </div>
        <div
          class="sc-gueYoa cHbybZ"
          style="margin-right: 16px;"
        >
          <span
            class="octicon octicon-comment"
            data-testid="icon"
            style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
          />
          <a
            class="sc-dmqHEX lmtyIK"
            href="https://api.github.com/gists/840386720b6a72d9e60dff8922f9747a/comments"
          >
            Comments
          </a>
        </div>
        <div
          class="sc-gueYoa cHbybZ"
        >
          <span
            class="octicon octicon-star"
            data-testid="icon"
            style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
          />
          <a
            class="sc-dmqHEX lmtyIK"
          >
            Stars
          </a>
        </div>
      </div>
    </div>
    <div
      class="sc-qRumB ftafra"
    >
      <p
        class="sc-beqWaB yDHQF"
      >
        <span>
          Created at: 
        </span>
        <span>
          2/31/2023
        </span>
      </p>
      <p
        class="sc-beqWaB yDHQF"
        style="margin-left: 16px;"
      >
        <span>
          Last updated: 
        </span>
        <span>
          2/31/2023
        </span>
      </p>
    </div>
    <p
      class="sc-kFuwaP hgxAcK"
    >
      test
    </p>
    <div
      class="sc-qRumB sc-jsMahE ftafra dTKjie"
    >
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
      <div
        class="sc-gueYoa cHbybZ"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-file-text"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-dmqHEX lmtyIK"
        />
      </div>
    </div>
  </div>
</div>
`;
